File: Programmers/신고결과받기.swift
import Foundation

func solution(_ id_list:[String], _ report:[String], _ k:Int) -> [Int] {
    var dic:[String : [String]] = [:]
    var idList:[String: Int] = [:]
    var arr:[Int] = Array(repeating: 0, count: id_list.count)
    
    for (idx, id) in id_list.enumerated() {
        idList[id] = idx
    }
    
    for val in report {
        let arr = val.split(separator: " ").map{ String($0) }
        let a = arr[0] //신고한 사람
        let b = arr[1] //신고당한 사람
        if dic[b] == nil {
            dic.updateValue([a], forKey: b)
        } else {
            if !dic[b]!.contains(a){
                dic[b]!.append(a)
            }
        }
    }
    
    for b in dic.keys {
        if dic[b]!.count >= k {
            // print(b)
            // print(dic[b]!)
            // print(idList)
            for n in dic[b]! {
                // print("--")
                // print(n)
                arr[idList[n]!] += 1
            }
            
        }
    }
    return arr
}

File: Programmers/신고결과받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        Map<String, HashSet<String>> reporterId = new HashMap<>();
        Map<String, Integer> reportedCnt = new HashMap<>();
        HashSet<String> reportSet = new HashSet<>(Arrays.asList(report));
        
        for(String info : reportSet) {
            String reporter = info.split(" ")[0];
            String reported = info.split(" ")[1];
            if (reporterId.containsKey(reporter)) {
                reporterId.get(reporter).add(reported);
            } else {
                reporterId.put(reporter, new HashSet<String>(){{
                    add(reported);
                }});
            }
            reportedCnt.put(reported, reportedCnt.getOrDefault(reported, 0)+1);
        }
        
        for (String reported : reportedCnt.keySet()){
            int reportedCount = reportedCnt.get(reported);
            if(reportedCount >= k){
                // 메일 발송 대상
                for(int i=0; i<id_list.length; i++){
                    if(reporterId.containsKey(id_list[i]) && reporterId.get(id_list[i]).contains(reported)) {
                        answer[i]++;
                    }
                }
            }
        }
        return answer;
    }
}